dev-resources/logback.xml

```diff
@@ -5,11 +5,11 @@
         </encoder>
     </appender>
 
-    <logger name="puppetlabs" level="error"/>
+    <logger name="puppetlabs" level="trace"/>
```

Comment:
> all tests would need to be wrapped in a with-test-logging call

yeah, that would be unacceptable.  I think we should be able to solve this logging issue the same way we've solved it in other repos; I don't exactly understand but it seems like the lack of a split b/w dev + test logback configs could solve it.